docs: add example for multi provider (#29909)
angular/angular: aio/content/guide/dependency-injection-providers.md

{@a factory-provider}
{@a factory-providers}

> ### Multiple providers

Reviewer: This should be moved as a sub-section under `Predefined tokens and multiple providers`

> You can also create multiple provider, Angular uses lot of multiple providers like `HTTP_INTERCEPTORS` and provides it to be used in applications. The advantage is you can add values into an array and use them whenever you need it, a simple example is, in case you want a mechanism to register external components within your framework, you can get those values via a multi provider and load them.

Reviewer 2: We should call them `multi-providers` rather than `multiple providers`. It's also not clear to me from this how they are different from normal providers and why they're benefiical.   Something like:

```suggestion
You can also create multi-providers.  Multi-providers are like normal providers, except that you can provide many values for a particular token. When the token is injected, you will receive an array of all the values that were provided (they don't override each other). This can be useful if... 
```

> You can also create multiple provider, Angular uses lot of multiple providers like `HTTP_INTERCEPTORS` and provides it to be used in applications. The advantage is you can add values into an array and use them whenever you need it, a simple example is, in case you want an mechanism to register external components within your framework, you can get those values via a multi provider and load them.

Reviewer 3: ```suggestion
You can also create multiple provider, Angular uses lot of multiple providers like `HTTP_INTERCEPTORS` and provides it to be used in applications. The advantage is you can add values into an array and use them whenever you need it, a simple example is, in case you want a mechanism to register external components within your framework, you can get those values via a multi provider and load them.
```

> The only change you have to make is to convert the above provider to multi provider is.

Reviewer 2: Let's make this more concrete

```suggestion
To convert the above provider to a multi-provider, add the `multi` flag:
```

<code-example path="dependency-injection/src/app/app.config.ts" region="multitoken" header="src/app/app.config.ts" linenums="false">
</code-example>

> To register the multi providers use the below syntax.

Reviewer 2: ```suggestion
To register the multi providers, use the below syntax.
```
